phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,205 @@
+            var messageEndIndex = unprocessed.indexOf('}\n\n');
+
+            if (messageEndIndex !== -1) {
+                var endOfMessageIndex = messageEndIndex + 3; // 3 is the length of "}\n\n"
+                var message = unprocessed.substring(0, endOfMessageIndex);
+                parse_message(message);
+                nextReadPosition += endOfMessageIndex;
+            }
+        } while (messageEndIndex !== -1);
+
+        if (xhReq.readyState === 4) {
+            $('#loading_indicator').css('display', 'none');
+            reset_polling();
+        }
+    }
+
+    function parse_message(messageJSON) {
+        $('#loading_indicator').css('display', 'none');
+
+        messageJSON = messageJSON.trim();
+        var responseObject = JSON.parse(messageJSON);
+
+        // Parse object
+        if (responseObject.hasOwnProperty('errors')) {
+            add_message('error', responseObject.errors)
+        }
+
+        if (responseObject.hasOwnProperty('warnings')) {
+            add_message('warning', responseObject.warnings)
+        }
+
+        if (responseObject.hasOwnProperty('logs')) {
+            add_message('log', responseObject.logs);
+        }
+
+        if (responseObject.hasOwnProperty('form')) {
+            add_form(responseObject.form);
+        }
+    }
+
+    function add_message(type, messages) {
+        // Get message containers
+        var errorContainer = $('#error-container');
+        var warningContainer = $('#warning-container');
+        var logContainer = $('#log-container');
+
+        var title, description, msgElement, arraySize = messages.length;
+        for (var i = 0; i < arraySize; i++) {
+            msgElement = $('<div />');
+            title = $(document.createElement('strong'));
+            title.text(messages[i].title);
+            msgElement.append(title);
+
+            if (messages[i].hasOwnProperty('description')) {
+                description = $(document.createElement('p'));
+                description.text(messages[i].description);
+                msgElement.append(description);
+            }
+
+            switch (type) {
+                case 'error':
+                    msgElement.addClass('errorbox');
+                    errorContainer.append(msgElement);
+                    break;
+                case 'warning':
+                    msgElement.addClass('warningbox');
+                    warningContainer.append(msgElement);
+                    break;
+                case 'log':
+                    msgElement.addClass('log');
+                    logContainer.append(msgElement);
+                    break;
+            }
+        }
+    }
+
+    function add_form(formHtml) {
+        var formContainer = $('#content-container');
+        formContainer.html(formHtml);
+        var form = $('#install_install');
+        intercept_form_submit(form);
+    }
+
+    function start_polling(xhReq) {
+        reset_polling();
+        pollTimer = setInterval(function () {
```

Comment:
That would require rewriting the PHP stuff, which is a possibility, but I wanted to use streaming response because by doing it that way I can avoid a lot of autoloading overhead and what not.